### DISCUSSION TITLE
Can the discriminator be automatically serialized?

### DISCUSSION THREAD

**Discussion by braincore** [pydantic/pydantic#8776] [Question]
The discriminator works correctly when deserializing an input using `model_validate()`. However, when serializing via `model_dump()` / `model_dump_json()`, the discriminator is absent. I'm able to achieve the desired outcome by specifying `WrapSerializer`, but I was wondering whether there was an automated method given the discriminator is known.

Example below:

```
class Metadata(BaseModel):
    name: str

class File(Metadata):
    size: int

class Folder(Metadata):
    pass

class FolderList(BaseModel):
    entries: list[
        Annotated[
            Union[
                Annotated[File, Tag("file")],
                Annotated[Folder, Tag("folder")],
            ],
            Discriminator(get_discriminator_value),
            WrapSerializer(ser_wrap, when_used='always'),
        ]
    ]
```

**Comment by GDYendell** [comment]
What are `get_discriminator_value` and `ser_wrap` in the example snippet? Could you give an example of the output you get from `model_dump()`?
